### DISCUSSION TITLE
Force charge when rates are negative

### DISCUSSION THREAD

**Discussion by Bart39** [springfall2008/batpred#1667] [Q&A]
Apologies if I have overlooked something but is there a way to configure so that if the electricity rate is negative charging is always planned ?

**Comment by gcoan** [comment]
In short no.  Predbat doesn't work like that with rate "thresholds" to determine whether to charge or discharge.

Predbat works on a cost optimisation so it will plan your battery charge and discharge in an optimal way to minimise your electricity cost (i.e. reduce import and maximise export).

Having said that, in many cases it charges your battery in the cheapest slots, letting the battery discharge due to house load or force export when rates are high, and discharging when rates are low. So negative charge rates should generally correspond to charging your battery.

If you don't like the plan predbat produces you can always override its behaviour by setting the times in select.predbat_manual_ force_idle/charge/export etc but its aiming to optimise your cost so in most cases this won't be necessary

**Reply by Bart39** [reply]
Thanks for the reply, that is what I thought.
However to counter your arguments, if the goal is cost optimisation then should it not always plan to charge the battery if the rate is negative (or at least freeze discharge) as it is more cost effective to use the grid and get paid for it rather than missing the opportunity and paying the grid later.
I will try to get some screenshots of the plan next time I get some negative rates to show what I mean (where I could have charged at a negative rate, but the plan choose to skip those slots and charge at a higher rate)

**Reply by gcoan** [reply]
If you find that Predbat isn't charging when it should do then might be worth raising this as a github issue, with log files, settings etc.

Last weekend was negative rates for me and Predbat was very busy, importing and exporting repeatedly, taking full advantage of the rates I was getting.  So might be that one of your settings is different from mine.

But happy to look at screenshots here as well
